apiserver/facades/client/modelconfig/modelconfig.go

```diff
@@ -285,3 +290,34 @@ func (c *ModelConfigAPI) Sequences(ctx context.Context) (params.ModelSequencesRe
 	result.Sequences = values
 	return result, nil
 }
+
+// GetModelSecretBackend isn't implemented in the ModelConfigAPIV3 facade.
+func (s *ModelConfigAPIV3) GetModelSecretBackend(context.Context) {}
+
+// GetModelSecretBackend returns the secret backend for the model.
+func (s *ModelConfigAPI) GetModelSecretBackend(ctx context.Context) (params.StringResult, error) {
+	result := params.StringResult{}
+	if err := s.auth.HasPermission(permission.ReadAccess, names.NewModelTag(s.modelUUID)); err != nil {
+		return result, errors.Trace(err)
+	}
+
+	name, err := s.configService.GetModelSecretBackend(ctx, coremodel.UUID(s.modelUUID))
```

Comment:
The model config service shouldn't be doing this work. It has nothing to do with the model config service. Separation of concerns.